### PR TITLE
Show per field errors

### PR DESCRIPTION
**Pull request by micgro42**
Bug: T308660

### CODE REVIEW

File: src/store/actions.ts [hunk 1]
```diff
@@ -50,25 +52,85 @@ export type InitParams = {
...
+			}
+			if ( state.language ) {
+				if ( !state.languageCodeFromLanguageItem && !state.spellingVariant ) {
+					commit(
+						ADD_PER_FIELD_ERROR,
+						{ field: 'spellingVariantErrors', error: { messageKey: 'wikibaselexeme-newlexeme-error-no-spelling-variant' } },
+					);
+				} else {
+					formData.validSpellingVariant = state.spellingVariant ||
+						state.languageCodeFromLanguageItem as string;
```
Comment (micgro42) [inline src/store/actions.ts:104]: Not sure if it is possible to avoid that type cast here

File: tests/integration/NewLexemeForm.test.ts [hunk 2]
```diff
@@ -235,6 +235,10 @@ describe( 'NewLexemeForm', () => {
 			await wrapper.trigger( 'submit' );
 
 			expect( createLexeme ).toHaveBeenCalledWith( 'foo', 'de', 'Q123', 'Q456' );
+
+			await nextTick();
+			await nextTick();
```
Comment (micgro42) [inline tests/integration/NewLexemeForm.test.ts:240]: Not sure if that should also be `flushPromises()`

Comment (lucaswerkmeister) [reply tests/integration/NewLexemeForm.test.ts:240]: probably… if this is between expectations on `createLexeme` and `goToTitle`, it’s not really about DOM modification, right?

File: src/store/actions.ts [hunk 3]
```diff
@@ -50,25 +52,85 @@ export type InitParams = {
...
 	lexemeCreator: LexemeCreator,
 	langCodeRetriever: LangCodeRetriever,
 	languageCodesProvider: LanguageCodesProvider,
 	tracker: Tracker,
 ): RootActions {
 	return {
-		async [ CREATE_LEXEME ]( { state, commit }: RootContext ): Promise<string> {
-			if ( !state.language || !state.lexicalCategory ) {
-				throw new Error( 'No language or lexical category!' ); // TODO
+		[ GET_VALID_INPUTS ]( { state, commit }: RootContext ): ValidLexemeData {
```
Comment (lucaswerkmeister) [inline src/store/actions.ts:69]: the name “get” sounds like it should be a getter to me (though I understand why it must be an action – it commits mutations)… can we find an alternative verb? maybe “assemble”?

File: src/store/actions.ts (hunk 1 above)
Comment (lucaswerkmeister) [reply src/store/actions.ts]: Pushed a commit avoiding it, feel free to revert if you don’t like i.

File: src/store/actions.ts (hunk 3 above)
Comment (micgro42) [reply src/store/actions.ts:69]: Done in that other PR that will be uploaded soon™

File: src/store/actions.ts [hunk 4]
```diff
@@ -50,25 +52,86 @@ export type InitParams = {
...
+			const formData: Partial<ValidLexemeData> = {};
+
+			if ( !state.lemma ) {
+				commit(
+					ADD_PER_FIELD_ERROR,
+					{ field: 'lemmaErrors', error: { messageKey: 'wikibaselexeme-newlexeme-error-no-lemma' } },
```
[the remark this answers is not in the document]
Comment (micgro42) [reply src/store/actions.ts:75]: let's keep your names. I'm in the process of polishing my commits right now anyway and I'm adjusting the names as I go

**Comment by micgro42** [comment]
This is being replaced by a couple of smaller PRs

**Pull request closed without merging**